tap_google_analytics/client.py

```diff
@@ -158,8 +161,8 @@ def _make_request(self, method, url, params=None, data=None):
             response = self.session.request(method, url, headers=headers, params=params)
 
         error_message = _is_json(response) and response.json().get("error", {}).get("message")
-        if response.status_code == 400 and error_message:
-            raise Exception("400 Client Error: Bad Request, details: {}".format(error_message))
+        if 400 <= response.status_code < 500 and error_message and not should_retry(response):
+            raise Exception("{} Client Error: Bad Request, details: {}".format(response.status_code, error_message))
```

Comment:
Maybe remove the explicit `Bad Request` here? It's not necessarily that, not sure if that part of the message can be pulled out of the response generically.

Reply:
Done, can you look at it now?